kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableImpl.java

```diff
@@ -60,34 +64,60 @@ public static Table forPath(Engine engine, String path) {
    * @return an instance of {@link Table} representing the Delta table at the given path
    */
   public static Table forPath(Engine engine, String path, Clock clock) {
-    String resolvedPath;
+    return forPathWithTableId(engine, path, Optional.empty(), clock);
+  }
+
+  public static Table forPathWithTableId(Engine engine, String path, TableIdentifier tableId) {
+    return forPathWithTableId(engine, path, Optional.of(tableId), System::currentTimeMillis);
+  }
+
+  public static Table forPathWithTableId(
+      Engine engine, String path, Optional<TableIdentifier> tableIdOpt, Clock clock) {
     try {
-      resolvedPath =
+      final String resolvedPath =
           wrapEngineExceptionThrowsIO(
               () -> engine.getFileSystemClient().resolvePath(path), "Resolving path %s", path);
+      return new TableImpl(resolvedPath, tableIdOpt, clock);
     } catch (IOException io) {
       throw new UncheckedIOException(io);
     }
-    return new TableImpl(resolvedPath, clock);
   }
 
-  private final SnapshotManager snapshotManager;
+  ////////////////////////////////
+  // Instance Methods / Members //
+  ////////////////////////////////
+
   private final String tablePath;
+  private final Optional<TableIdentifier> tableIdOpt;
   private final Clock clock;
 
-  public TableImpl(String tablePath, Clock clock) {
+  private final Path dataPath;
+  private final Path logPath;
+  private final SnapshotManager snapshotManager;
+
+  private TableImpl(String tablePath, Optional<TableIdentifier> tableIdOpt, Clock clock) {
     this.tablePath = tablePath;
-    final Path dataPath = new Path(tablePath);
-    final Path logPath = new Path(dataPath, "_delta_log");
```

Comment:
we were duplicating this Path creation in the `getDataPath` and `getLogPath` methods. might as well just save them as instance variables.